pkg/pillar/cmd/volumemgr/handlecontent.go

```diff
@@ -189,13 +195,60 @@ func createContentTreeStatus(ctx *volumemgrContext, config types.ContentTreeConf
 				}
 				publishBlobStatus(ctx, rootBlob)
 			}
+			AddBlobsToContentTreeStatus(ctx, status, strings.ToLower(config.ContentSha256))
 		}
 	}
 	publishContentTreeStatus(ctx, status)
 	log.Infof("createContentTreeStatus for %v Done", config.ContentID)
 	return status
 }
 
+//AddBlobsToContentTreeStatus adds blob to ContentTreeStatus.Blobs also increments RefCount of the respective BlobStatus.
+//NOTE: This should be the only method to add blobs into ContentTreeStatus.Blobs
+func AddBlobsToContentTreeStatus(ctx *volumemgrContext, status *types.ContentTreeStatus, blobShas ...string) error {
+	log.Infof("AddBlobsToContentTreeStatus(%s): for blobs %v", status.ContentID, blobShas)
+	for _, blobSha := range blobShas {
+		blobStatus := lookupBlobStatus(ctx, blobSha)
+		if blobStatus == nil {
+			err := fmt.Errorf("AddBlobsToContentTreeStatus(%s): No BlobStatus found for blobHash: %s",
+				status.ContentID.String(), blobSha)
+			log.Errorf(err.Error())
+			return err
+		}
+		status.Blobs = append(status.Blobs, blobSha)
```

Comment:
How about adding a comment here to the effect that
It is critical that the append to Blobs and incrementing of the refcount are atomic since we depend on that for decrementing the refcount when the ContentTreeStatus is deleted.

Reply:
And also comment
If the same sha appears in multiple places in the ContentTree we intentionally add it twice to the list of Blobs so that we can have two reference counts on that blob.

Reply:
Added.

Reply:
I don't see "If the same sha appears in multiple places in the ContentTree we intentionally add it twice to the list of Blobs so that we can have two reference counts on that blob." in the diff

Reply:
Added